### PR TITLE
Added workflow compatibility

### PR DESCRIPTION
**Pull request by cjbush**

### CODE REVIEW

**Comment by cjbush** [comment]
Looks like the failure is because MSTest depends on this plugin and it didn't update to use the snapshot. How should I handle that?

**Comment by nilleb** [comment]
Hello CJ,

http://repo.jenkins-ci.org/public/org/jenkins-ci/plugins/emma/

The version 1.30-SNAPSHOT is not present on this repository.
It is maybe available on some other repository. I am not a Java
professional, so I can't really help you. If I were you, I would have a
look at the articles pointed by this stackoverflow post:
http://stackoverflow.com/a/1368663 ;-)

And, by the way, thank you very much for your code! :-)

**Comment by cjbush** [comment]
I believe the current version of the plugin is 1.29, so I was the one who set it to 1.30-SNAPSHOT as a development prerelease.

**Comment by nilleb** [comment]
That's it. But if you really want to use a development prerelase, you could
be able to find it on some other repository, definitely not on the official
jenkins-ci repo.

**Comment by cjbush** [comment]
I guess my question is what would you like me to set the version to for the build to pass?

**Comment by nilleb** [comment]
1.29, unless you really need the 1.30-SNAPSHOT. If you really want the
1.30-SNAPSHOT, it's up to you to modify the pom.xml to instruct mvn about
where to find it!

But that's not a question of 'me', that's the Java build system (mvn) ! :-)

Le mar. 26 avr. 2016 à 21:38, Chris Bush <notifications@github.com> a
écrit :

**Comment by cjbush** [comment]
OK, the problem is that this change depends on a change I also made to the Emma plugin. That change is currently sitting in a merge request. I guess this one will have to wait until that one is accepted and released.

**Comment by nilleb** [comment]
You're right, Chris. Good luck!

Le mar. 26 avr. 2016 à 22:11, Chris Bush <notifications@github.com> a
écrit :

File: pom.xml [hunk 1]
```diff
@@ -90,6 +90,12 @@
             <version>1.4</version>
             <type>jar</type>
         </dependency>
+        <dependency>
```
Comment (Aeon512) [inline pom.xml]: Remove this dependency completely as it's no longer needed

**Comment by Aeon512** [comment]
Is there any test-case for this pipeline functionality available? or some description how to call this step?

File: pom.xml (hunk 1 above)
Comment (nilleb) [inline pom.xml]: What will happen when the PR will be built on the Cloudbees CI server, which hasn't the emma-plugin 1.30-snapshot?

Comment (cjbush) [inline pom.xml]: That will have to be changed to 1.30 when the PR on the Emma plugin project gets accepted and released. We're currently trying to get that PR accepted.

**Comment by cjbush** [comment]
@Aeon512 I didn't write a test case for it. I'm using it on my Jenkins server currently by doing this generic build step:

step($class: 'MSTestPublisher', testResultsFile: 'Results.trx', enableCodeCoverageAnalysis: true)

Not ideal, but I wasn't sure how to create an alias with the new API. I'm still learning how it all works.

**Comment by stevemac007** [comment]
Any thoughts on progressing this?

**Comment by b-szypelow** [comment]
Any update?

**Comment by nilleb** [comment]
There's still no emma-1.30 (and no 1.30-snapshot).

@cjbush doesn't look to be interested anymore. I am not willing anymore to maintain this project.
The code is here, available for anyone who would be willing to take it. The official Jenkins plugin contributor documentation should be sufficient to [build it](https://wiki.jenkins-ci.org/display/JENKINS/Plugin+tutorial#Plugintutorial-BuildingaPlugin) (and produce a jar who will let you install it on a Jenkins instance).

**Comment by casz** [comment]
closing as plugin no longer uses emma and works with pipeline plugin

**Pull request closed without merging**